Source/PluginProcessor.cpp

```diff
@@ -130,21 +124,94 @@ void SoomplerAudioProcessor::processBlock (AudioBuffer<float>& buffer, MidiBuffe
         return;
     }
 
-    auto midiChannelBuffer = filterMidiMessagesForChannel(midiMessages, 1);
+    // TODO maybe this all multi-bus thing is unnesesary?
+//    auto busCount = getBusCount(false);
+//    for (auto busNr = 0; busNr < busCount; ++busNr) {
+//        auto midiChannelBuffer = filterMidiMessagesForChannel(MidiBuffer, busNr + 1);
+//        auto audioBusBuffer = getBusBuffer(buffer, false, busNr);
+
+//        synth.renderNextBlock();
+//    }
+
+    lastMidiEvents = filterMidiMessagesForChannel(midiMessages, 1);
     auto audioBusBuffer = getBusBuffer(buffer, false, 0);
 
-    synth.renderNextBlock(audioBusBuffer, midiChannelBuffer, 0, audioBusBuffer.getNumSamples());
+    synth.renderNextBlock(audioBusBuffer, lastMidiEvents, 0, audioBusBuffer.getNumSamples());
 }
 
 void SoomplerAudioProcessor::processTransport(AudioBuffer<float>& buffer)
 {
-    AudioSourceChannelInfo audioSource{};
+    AudioSourceChannelInfo audioSource;
     audioSource.buffer = &buffer;
     audioSource.startSample = 0;
     audioSource.numSamples = buffer.getNumSamples();
     transportSource.getNextAudioBlock(audioSource);
 }
 
+void SoomplerAudioProcessor::setSampleStartPosition(int64 sample)
+{
+    startSample = sample;
+    transportSource.setNextReadPosition(startSample);
+
+    auto voice = static_cast<soompler::ExtendedVoice*> (synth.getVoice(0));

```

Comment:
synth.getVoice(0) always not null? Are you sure?

Reply:
dynamic cast

Reply:
please, write some arguments to use dynamic instead of static.

Reply:
Voice is null, when someone forgot to place addition of voice in constructor. I think this is critical, so error produced in this block of code should be indicator. No need to overcheck here. We don't expect voice to be null in any normal circumstances.